pkg/securitypolicy/securitypolicyenforcer.go

```diff
@@ -483,17 +381,70 @@ func equalForOverlay(a1 []string, a2 []string) bool {
 	return true
 }
 
-func possibleIndicesForID(containerID string, mapping map[int]map[string]struct{}) []int {
-	var possibles []int
-	for index, ids := range mapping {
+func (se *StandardSecurityPolicyEnforcer) possibleIndicesForID(containerID string) []int {
+	var possibleIndices []int
+	for index, ids := range se.ContainerIndexToContainerIds {
 		for id := range ids {
 			if containerID == id {
-				possibles = append(possibles, index)
+				possibleIndices = append(possibleIndices, index)
 			}
 		}
 	}
+	return possibleIndices
+}
 
-	return possibles
+// EnforceMountPolicy for StandardSecurityPolicyEnforcer validates various
+// default mounts injected into container spec by GCS or containerD
+func (se *StandardSecurityPolicyEnforcer) EnforceMountPolicy(sandboxID, containerID string, spec *oci.Spec) (err error) {
+	defaultAllowedMounts, err := getDefaultMountConstraints(sandboxID, spec)
+	if err != nil {
+		return err
+	}
+	possibleIndices := se.possibleIndicesForID(containerID)
```

Comment:
I think we probably need to add some locking around `ContainerIndexToContainerIds` to protect against concurrent operations. Maybe other state like `Devices` as well?

Reply:
I suspect this goes beyond just the mount enforcement, but haven't looked closely.

Reply:
There's a mutex for the entire object. It should be used but isnt being used in this new method. Excellent catch.

Example usage:

https://github.com/microsoft/hcsshim/blob/2e63427f698b1391b5c94377206c24f8399d568f/pkg/securitypolicy/securitypolicyenforcer.go#L184

Reply:
done. honestly surprised I missed this...